modules/apps/commerce/commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/internal/messaging/CommerceOrderStatusMessageListener.java

```diff
@@ -52,8 +51,7 @@ public class CommerceOrderStatusMessageListener extends BaseMessageListener {
 
 	@Override
 	protected void doReceive(Message message) throws Exception {
-		JSONObject jsonObject = _jsonFactory.createJSONObject(
-			String.valueOf(message.getPayload()));
+		JSONObject jsonObject = (JSONObject)message.getPayload();
 
```

Comment:
This is kind of a shot in the dark, but from debugging at least doing a cast seems to go through a lot less methods calls then using _jsonFactory to create a jsonObect.